provider/lib.go

```diff
@@ -142,3 +142,30 @@ func MultilineGrep(ctx context.Context, window int, path, pattern string) (int,
 
 	return -1, scanner.Err()
 }
+
+// GetIncludedPathsFromConfig returns validated includedPaths from provider settings
+// if allowFilePaths is not set, path to a file is converted into a path to its base dir
```

Comment:
Can you walk me through the use case for this option to be false?

Reply:
@shawn-hurley this doesn't affect users, this is for providers...so with Java, we can only scope the path upto granularity of a package (as opposed to a class or file). So if user inputs `/util/File.java` we can only filter upto `/util`. I thought this would be potentially useful elsewhere, but I can push this specific logic into java provider too keeping this lib function generic